lightning/src/ln/channel.rs

```diff
@@ -3131,8 +3131,15 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 	}
 
 	/// Allowed in any state (including after shutdown)
-	pub fn get_announced_htlc_max_msat(&self) -> Option<u64> {
-		self.our_htlc_maximum_msat
+	pub fn get_announced_htlc_max_msat(&self) -> u64 {
+		let mut result = cmp::min(
+			self.channel_value_satoshis * 9 / 10, // upper bound by capacity
+			Channel::<ChanSigner>::get_our_max_htlc_value_in_flight_msat(self.channel_value_satoshis)
+		);
```

Comment:
@TheBlueMatt why not just max_inflight? the 9/10 seems potentially confusing, so just want to make sure it's actually necessary...

Reply:
I think this is just a heuristic to reduce the failure rate in case if the channel was previously used to route a few sats...
It helps with both malicious and (very trivial) non-malicious cases.

If I understand it correctly, I do think it's a useful bound.
I'm not sure for whom this would be confusing, maybe you have a better intuition of lightning users/operators?

Reply:
I guess I just meant that I was confused by it, lol.

OK, seems like that'd be worth documenting. I'd add a comment along the lines of -- 
"Upper bound our max_htlc by 9/10 of the channel capacity. This protects against the following attack:
If the channel has already routed a few sats, then without this upper bound the attacker may be able to DoS our node by repeatedly attempting and failing to route `our_channel_capacity - 1` to us. Whereas, if we advertise our max_htlc as 9/10 of channel capacity, no one will try to route `our_channel_capacity_amount` via our channels, so an attacker would at least have to pay a nontrivial amount to accomplish the same DoS attack."

Reply:
I actually meant an attacker routing those fews sats in the first place, not failing to route... As you see, the scope may be broad here. As well as non-malicious cases when few sats were routed by honest users.
I would add one sentence, something about amount "slack", but not sure how to word it better.

Reply:
Gotcha -- I'd like to be super explicit about what exactly we're protecting against, to be well-documented for future developers. Let me know what phrasing you think would be best :)